plugins/parsers/csv/parser.go

```diff
@@ -197,23 +198,30 @@ func (p *Parser) Parse(buf []byte) ([]telegraf.Metric, error) {
 	}
 
 	r := bytes.NewReader(buf)
-	return parseCSV(p, r)
+	metrics, err := parseCSV(p, r)
+	if err != nil && errors.Is(err, io.EOF) {
+		return nil, parsers.ErrNotEnoughData
```

Comment:
I'm not sure I understand the new parsers.ErrNotEnoughData error. Doesn't io.EOF conceptually also mean there isn't enough data? Is the difference that the new error is returned when the parser expects more data eventually and EOF is final with no chance of more data?

Reply:
The idea was to not reuse the existing error to minimize side-effects for non-CSV parsers that also might return `io.EOF`. By using a new error and only apply it to CSV we can be sure that the previous `if` condition is still there...